pystac/link.py

```diff
@@ -183,10 +183,15 @@ def get_href(self, transform_href: bool = True) -> Optional[str]:
                 *pystac.EXTENSION_HOOKS.get_extended_object_links(self.owner),
             ]
             # if a hierarchical link with an owner and root, and relative catalog
-            if root and root.is_relative() and self.rel in rel_links:
-                owner_href = self.owner.get_self_href()
-                if owner_href is not None:
-                    href = make_relative_href(href, owner_href)
+            if root and root.is_relative():
+                if self.rel in rel_links:
+                    owner_href = self.owner.get_self_href()
+                    if owner_href is not None:
+                        href = make_relative_href(href, owner_href)
+                elif self.target in root.get_target_hierarchy():
+                    owner_href = self.owner.get_self_href()
+                    if owner_href is not None:
+                        href = make_relative_href(href, owner_href)
```

Comment:
What do you think about this refactor?

```suggestion
                if self.rel in rel_links or root.has_descendant(self.target):
                    owner_href = self.owner.get_self_href()
                    if owner_href is not None:
                        href = make_relative_href(href, owner_href)
```

w/ the necessary changes to add a `has_descendant` method to `Catalog` in place of `get_target_hierarchy`? This would read a bit cleaner to me at least.

Reply:
Oops.  Yeah.  Overthought this.  `or` short circuits.  Duh.